test/queries/queries-good.sql

```diff
@@ -18,7 +18,35 @@ SELECT * FROM t WHERE a = ? AND b = ?;
 SELECT City.name, Product.category, SUM(price) FROM fact INNER JOIN City ON fact.city_id = City.id INNER JOIN Product ON fact.product_id = Product.id GROUP BY City.name, Product.category;
 SELECT SUBSTR(a, 3, 5) FROM t;
 SELECT * FROM t WHERE a = DATE '1996-12-31';
-# JOIN
+SELECT 1 UNION SELECT 1;
```

Comment:
You added a lot of things that I would love to see tested more properly. Could you please add more complex examples with actual subqueries? I would also be interested in some edge cases that are not on the happy path, e.g., in `queries-bad.sql`. I would also encourage you to think of a thorough test case in `select_tests.cpp`

Reply:
Good catch, I actually broke union-in-subquery syntax. I added a new commit that addresses it.